shared/js/initApp.js

```diff
@@ -150,7 +150,43 @@ $(window).on('function_editor_closed', function() {
   }
 })(appOptions.level);
 
+/**
+ * Only execute the given argument if it is a function.
+ * @param callback
+ */
+function callbackSafe(callback, data) {
```

Comment:
It's only needed here because `callback` is an Event object if the handler is bound as:
```javascript
$('#runButton').click(dashboard.saveProject);
```
An alternative would be to explicitly pass in an empty function, then this method wouldn't be needed:
```javascript
$('#runButton').click(function () {
  dashboard.saveProject(function () {});
});
```